Tools/Scripts/import-source-map-tests

```diff
@@ -0,0 +1,106 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+# OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+# TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+# Updates the source map spec tests in LayoutTests/tc39-tg4/source-map-tests
+# with the latest upstream tests.
+
+import argparse
+import pathlib
+import re
+import shutil
+import sys
+
+from webkitpy.common.host import Host
+from webkitpy.common.system.filesystem import FileSystem
+from webkitpy.common.webkit_finder import WebKitFinder
+from webkitpy.common.checkout.scm.git import Git
+
+host = Host()
+host.initialize_scm()
+port = host.port_factory.get()
+layout_tests_directory = port.layout_tests_dir()
+source_map_repo_url = "https://github.com/tc39/source-map-tests.git"
+source_map_repo_branch = "main"
+
+def generate_readme_contents(git_hash):
+    return """\
+This directory contains the imported TG4 source map tests
+
+  - URL: {url}
+  - Commit: {git_hash}
+
+To update the tests, run the script at Tools/Scripts/import-source-map-tests
+with no arguments. This will checkout out the repo above and copy the new
+tests over. You can also specify a different repo and branch (see usage
+info with the -h flag).
+""".format(url=source_map_repo_url, git_hash=git_hash)
+
+def checkout_tests(continuation):
+    with host.filesystem.mkdtemp() as temporary_directory:
+        checkout_folder = "source-map-tests"
+        checkout_directory = host.filesystem.join(temporary_directory, checkout_folder)
+        print("Checking out tests from " + source_map_repo_url + " into " + checkout_directory)
+        Git.clone(source_map_repo_url, checkout_directory, host.executive)
+        git = Git(checkout_directory, None, executive=host.executive, filesystem=host.filesystem)
+        git.checkout(source_map_repo_branch)
+        git_hash = git.native_revision(checkout_directory)
+        print("Checked out git hash " + git_hash)
+        continuation(pathlib.Path(checkout_directory), git_hash)
+
+def copy_tests(tmp_path, git_hash):
+    tests_path = host.filesystem.join(layout_tests_directory, "imported", "tc39-tg4", "source-map-tests")
+    git = Git(tests_path, None, executive=host.executive, filesystem=host.filesystem)
+    status_output = git.run([git.executable_name, "status", tests_path]).splitlines()
+    test_dir_clean = re.search("working tree clean", status_output[-1] if status_output else "")
```

Comment:
Basically anything parsing the output of a git command is an anti-pattern; it would be better to do `git status --porcelain=v2 -z` and then check that the output is empty.